### ISSUE TITLE
Tracking (aws-ecs-patterns): Are there any new patterns that users would like to see?

### ISSUE THREAD

**Issue by piradeepk** [aws/aws-cdk#5482]
Please comment with any new patterns that you would like to see (Specific to ECS). Please provide the following in each comment:

-----
##### Use Case

why do you need this feature/what would it be used for(scenarios)?

##### Proposed Solution

If you have a solution proposal

##### Other

Any additional information

**Comment by nathanpeck** [comment]
I want an AWS App Mesh enabled ECS service as a pattern: https://github.com/aws/aws-cdk/issues/4566

**Comment by nathanpeck** [comment]
I also want a REST API pattern, where multiple services can be mounted onto different paths on an ALB like:

/api/users -> users service
/api/orders -> order service

**Comment by pgarbe** [comment]
I'd like to have a resilient scheduled Fargate task as described here: https://cloudonaut.io/resilient-task-scheduling-with-ecs-fargate-cron-scheduled-task/ 

tldr; it uses Step Functions instead of CloudWatch Rules

**Comment by misterjoshua** [comment]
**Use Case**
I'd like to use a pre-existing load balancer listener for multiple ECS services that are differentiated by ALB rules and rule priorities.

**Proposed Solution**
Add a `listener` prop to `ApplicationLoadBalancedServiceBaseProps` which identifies which load balancer listener rules should be added to and an `addListenerRule(rule: ApplicationTargetProps)` method to `ApplicationLoadBalancedEc2Service` and `ApplicationLoadBalancedFargateService`. When addListenerRule() is called, the pattern construct would add `ApplicationListenerRule`s that reference the `listener` specified in the constructor props.

```ts
const loadBalancedEcsService = new ecsPatterns.ApplicationLoadBalancedEc2Service(stack, 'Service', {
  cluster,
  // New prop:
  listener: albListener,
  memoryLimitMiB: 1024,
  taskImageOptions: {
    image: ecs.ContainerImage.fromRegistry('test'),
  },
});

// New method:
loadBalancedEcsService.addListenerRule({
  containerPort: 80,
  hostHeader: "something.example.com",
  priority: 10,
});
```

**Comment by SoManyHs** [comment]
We are now leveraging ecs service extensions to enable new patterns. Closing.
